apps/callscreen/js/handled_call.js

```diff
@@ -292,29 +274,31 @@ HandledCall.prototype.formatPhoneNumber =
 
 HandledCall.prototype.replacePhoneNumber =
   function hc_replacePhoneNumber(phoneNumber, ellipsisSide) {
-    this.numberNode.textContent = phoneNumber;
-    this.formatPhoneNumber(ellipsisSide);
+    if (phoneNumber.id) {
+      navigator.mozL10n.setAttributes(this.numberNode, phoneNumber.id);
+      this.numberNode.style.fontSize = '';
+    } else {
+      this.numberNode.removeAttribute('data-l10n-id');
+      this.numberNode.textContent = phoneNumber;
+      this.formatPhoneNumber(ellipsisSide);
+    }
+
+    this._cachedInfo = phoneNumber;
 };
 
 HandledCall.prototype.restorePhoneNumber =
   function hc_restorePhoneNumber() {
-    this.numberNode.textContent = this._cachedInfo;
-    this.formatPhoneNumber('end');
+    this.replacePhoneNumber(this._cachedInfo, 'end');
 };
 
 HandledCall.prototype.updateDirection = function hc_updateDirection() {
-  var self = this;
   var classList = this.node.classList;
   if (this._initialState == 'incoming') {
     classList.add('incoming');
-    LazyL10n.get(function localized(_) {
-      self.node.setAttribute('aria-label', _('incoming'));
-    });
+    this.node.setAttribute('aria-label', 'incoming');
```

Comment:
how is this supposed to work? I think what you want here is

this.node.setAttribute('data-l10n-id', 'incoming');

and in your l10n resources:

<incoming
  ariaLabel: "Foo">